pkg/stub/catalogsourceconfig_handler.go

```diff
@@ -0,0 +1,77 @@
+package stub
+
+import (
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// createCatalogSource creates a new CatalogSource CR and all the resources it requires
+func createCatalogSource(cr *v1alpha1.CatalogSourceConfig) error {
+	// Create the ConfigMap that will be used by the CatalogSource
+	catalogConfigMap := newConfigMap(cr)
+	err := sdk.Create(catalogConfigMap)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		logrus.Errorf("Failed to create catalog source : %v", err)
+		return err
+	}
+
+	err = sdk.Create(newCatalogSource(cr, catalogConfigMap.Name))
+	if err != nil && !errors.IsAlreadyExists(err) {
+		logrus.Errorf("Failed to create catalog source : %v", err)
+		return err
+	}
+	return nil
+}
+
+// newCatalogSource returns a CatalogSource object
+func newCatalogSource(cr *v1alpha1.CatalogSourceConfig, configMapName string) *olm.CatalogSource {
+	name := "csc-cs-" + cr.Name
```

Comment:
Probably having a constant or var here for `csc-cs-` so that we can import it in other places.

Reply:
+1 
I already added this while writing the e2e tests. You will see this on the next update to this PR.